glassfish-on-suse/azuredeploy.json

```diff
@@ -0,0 +1,539 @@
+        "Standard_D11",
+        "Standard_D12",
+        "Standard_D13",
+        "Standard_D14",
+        "Standard_D2_v2",
+        "Standard_D3_v2",
+        "Standard_D4_v2",
+        "Standard_D5_v2",
+        "Standard_D11_v2",
+        "Standard_D12_v2",
+        "Standard_D13_v2",
+        "Standard_D14_v2",
+        "Standard_G1",
+        "Standard_G2",
+        "Standard_G3",
+        "Standard_G4",
+        "Standard_G5",
+        "Standard_DS2",
+        "Standard_DS3",
+        "Standard_DS4",
+        "Standard_DS11",
+        "Standard_DS12",
+        "Standard_DS13",
+        "Standard_DS14",
+        "Standard_DS2_v2",
+        "Standard_DS3_v2",
+        "Standard_DS4_v2",
+        "Standard_DS5_v2",
+        "Standard_DS11_v2",
+        "Standard_DS12_v2",
+        "Standard_DS13_v2",
+        "Standard_DS14_v2",
+        "Standard_GS1",
+        "Standard_GS2",
+        "Standard_GS3",
+        "Standard_GS4",
+        "Standard_GS5"
+      ],
+      "metadata": {
+        "description": "VM(s) size"
+      }
+    },
+    "osImage": {
+      "type": "string",
+      "allowedValues": [ "opensuse", "sles" ],
+      "defaultValue": "opensuse",
+      "metadata": {
+        "description": "Operating System to use for VM image(s)"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username"
+      }
+    },
+    "glassfishAdminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "GlassFish admin user password"
+      }
+    },
+    "sshPublicKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "SSH public key on all VMs"
+      }
+    }
+  },
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "apiVersionStorage": "2015-06-15",
+    "apiVersionCompute": "2015-06-15",
+    "apiVersionNetwork": "2015-06-15",
+    "apiVersionLinkTemplate": "2015-01-01",
+    "namePrefix": "[concat(toLower(parameters('vmPrefix')), '-')]",
+    "availabilitySetName": "[concat(variables('namePrefix'), 'nodes')]",
+    "subnetName": "[concat(variables('namePrefix'), 'subnet')]",
+    "nsgName": "[concat(variables('subnetName'), '-NSG')]",
+    "publicIPAddressID": "[resourceId('Microsoft.Network/publicIPAddresses',parameters('publicIPAddressName'))]",
+    "lbID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'))]",
+    "frontEndIPConfigID": "[concat(variables('lbID'),'/frontendIPConfigurations/LoadBalancerFrontEnd')]",
+    "lbPoolName": "lbBackend",
+    "lbPoolID": "[concat(variables('lbID'),'/backendAddressPools/', variables('lbPoolName'))]",
+    "lbProbeName": "[concat(variables('namePrefix'), 'tcpProbe')]",
+    "lbProbeID": "[concat(variables('lbID'),'/probes/', variables('lbProbeName'))]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',parameters('vnetName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/', variables ('subnetName'))]",
```

Comment:
simplify to: resourceId('Microsoft.Network/virtualNetworks/subnets', variables('vnetName'), variables('subnetname')